src/python/grpcio/grpc/_server.py

```diff
@@ -390,6 +398,16 @@ def unary_request():
     return unary_request
 
 
+def _get_default_initial_metadata_operation(state):
+    with state.condition:
+        if state.initial_metadata_allowed:
```

Comment:
Are these two checks necessary here?

Reply:
Not strictly speaking, no. But `state.condition` is an `RLock`, so it should be fine here. The alternative, in my opinion would be to add a `_locked` suffix to the function to indicate that it should only be called when the condition is held, but I don't like that option as much.

Reply:
Fair point.